Objects/genobject.c

```diff
@@ -1893,16 +1898,17 @@ async_gen_athrow_send(PyAsyncGenAThrow *o, PyObject *arg)
 
 yield_close:
     o->agt_gen->ag_running_async = 0;
+    o->agt_state = AWAITABLE_STATE_CLOSED;
```

Comment:
`ag_running_async` tracks whether there's some call to `asend`/`athrow` in progress. `agt_state` tracks whether *this* call to `asend`/`athrow` is in progress. So every time we set `ag_running_async = 0`, we should also set `agt_state = AWAITABLE_STATE_CLOSED`.